### PR TITLE
Fixes from virtual CICD workshop

### LINKED ISSUE #111: Docs(CICD): Add a note to assist user to fix below error when merging origin/cicd-ff branch
When user tries to merge the branch below, they receive this error because they forgot to uncheck `copy the main branch only` when forking the repo.

```
git merge origin/cicd-ff
```

Error Message:
```
merge: origin/cicd-ff - not something we can merge
```

Steps to fix the issue without re-forking the repo.  This is useful if running a virtual workshop and users are having issues.

```
git remote add upstream https://github.com/aristanetworks/ci-workshops-avd.git
git fetch upstream
git merge upstream/cicd-ff
```

### LINKED ISSUE #112: Make passphrase a copy block
Edit will make flow easier for students

### LINKED ISSUE #113: Docs(CI/CD): Add Steps#'s to each section in lab guide
For the CI/CD Lab Guide, let's add Step#'s for each section.  This way we can refer to where we are when running a lab and make sure attendees have caught to Step# ....

https://aristanetworks.github.io/avd-workshops/cicd-basics/

This is already done on the AVD Lab Guide...

https://aristanetworks.github.io/avd-workshops/avd-lab-guide/

### LINKED ISSUE #115: Docs(CI/CD): Breakup pre-commit codeblocks to separate command from output
Example:

```
➜  ci-workshops-avd git:(main) ✗ pre-commit run -a
trim trailing whitespace.................................................Passed
fix end of files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook
```

Break this into 2 code blocks like:

```
pre-commit run -a
```
Output:
```
trim trailing whitespace.................................................Passed
fix end of files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook
```

### PR DESCRIPTION
**Pull request by JulioPDX**
This PR corrects or implements the following:

- Fixes #111 
- Fixes #112 
- Fixes #113
- Fixes #115 
- Reorganize requirements files with pre-commit hook
- Update all versions of pre-commit actions
- Remove material-mkdocs-extensions, deprecated

### CODE REVIEW

File: workshops/cicd-basics.md
Comment (mthiel117) [inline workshops/cicd-basics.md:1]: I would make the AVD and CICD guides both match on bold or non-bold Step #'s

**Review by mthiel117 — COMMENTED** [review]
One small change requested.  Make both AVD and CICD guides have matching boldness of Step #'s.

File: workshops/cicd-basics.md
Comment (JulioPDX) [reply workshops/cicd-basics.md:1]: can do

**Review by mthiel117 — APPROVED** [review]
LGTM

**Pull request merged**